rpcs3/Emu/Cell/PPUTranslator.h

```diff
@@ -99,6 +101,14 @@ class PPUTranslator final : public cpu_translator
 		return SetVr(vr, expr.eval(m_ir));
 	}
 
+	llvm::Value* VecHandleNan(llvm::Value* val);
+
+	template <typename T>
+	auto vec_handle_nan(T&& expr)
+	{
+		return VecHandleNan(expr.eval(m_ir));
```

Comment:
maybe return a value_t<> instead of llvm::Value* for consistency.

Reply:
i considered it but `set_vr` above this is the same style and does _not_ return `value_t`, so the _current_ way is consistent with that

Reply:
I mean its consistent with other functions in CPUTranslator.h
It's also odd to mix "old style" llvm::Value* functions with the "new style" value_t<>.

Reply:
set_vr also doesn't return anything so it's not a fair comparison.

Reply:
ah yeah, does call `eval` but doesn't return, i'll change it

Reply:
wait i'm not crazy it does return lol
but ret type is `void` 🤔 
```
	template <typename T>
	void set_vr(u32 vr, T&& expr)
	{
		return SetVr(vr, expr.eval(m_ir));
	}
```

Reply:
i'll change it

Reply:
pushed